src/next/azext_next/custom.py

```diff
@@ -0,0 +1,119 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import json
+
+from knack.util import CLIError
+
+
+def _get_api_url():
+    return "https://cli-recommendation.azurewebsites.net/api/RecommendationService"
+
+
+def _get_last_cmd():
+    '''Get last executed command from local log files'''
+    import os
+    his_file_name = os.path.join(os.environ['HOME'], '.azure', 'recommendation', 'cmd_history.log')
+    with open(his_file_name, "r") as f:
+        lines = f.read().splitlines()
+        lines = [x for x in lines if x != 'next']
+        return lines[-1]
+    return ''
+
+
+def _update_last_cmd(cmd):
+    import os
+    his_file_name = os.path.join(os.environ['HOME'], '.azure', 'recommendation', 'cmd_history.log')
+    with open(his_file_name, "a") as f:
+        f.write("{}\n".format(cmd))
+
+
+def _get_recommend_from_api(last_cmd, request_type, top_num=5, extra_data=None):  # pylint: disable=unused-argument
+    '''query next command from web api'''
+    import requests
+    url = _get_api_url()
+    payload = {
+        "command": last_cmd,
+        "top_num": top_num
+    }
+    response = requests.post(url, json.dumps(payload))
+    if response.status_code != 200:
+        raise CLIError("Failed to connect to '{}' with status code '{}' and reason '{}'".format(
+            url, response.status_code, response.reason))
+    recommends = response.json()['data']
+
+    return recommends
+
+
+def _read_int(msg, default_value=0):
+    ret = input(msg)
+    if ret == '' or not ret.isnumeric():
+        ret = default_value
+    else:
+        ret = int(ret)
+    return ret
+
+
+def _give_recommends(recommends):
+    print("")
+    idx = 0
+    for rec in recommends:
+        if 'reason' in rec:
+            reason = rec['reason']
+        else:
+            reason = " people use this command in next step."
+        idx += 1
+        print("{}. az {}".format(idx, rec['command']))
+        print("Recommended reason: {:.1f}% {}".format(rec['ratio'] * 100, reason))
```

Comment:
We may also need to consider the case where `rec['ratio']` does not exist
```suggestion
        if rec['ratio']:
               reason = "{:.1f}% {}".format(rec['ratio'] * 100, reason)
        print("Recommended reason: {}".format(reason))
```